controllers/etcd_controller.go

```diff
@@ -587,6 +644,17 @@ func (r *EtcdReconciler) reconcileStatefulSet(cm *corev1.ConfigMap, svc *corev1.
 			return nil, err
 		}
 
+		// delete PVC on etcd scale down based on volumeRetentionPolicy
+		if handleScaleDown {
```

Comment:
Can we find a way to modularize to avoid repetition [here](https://github.com/gardener/etcd-druid/pull/59/files#diff-8e5c816f8b2f1adeef36a64d81b3e0f6R300-R308)?

Reply:
The deletion and scale down cases have not really been modularized. Only the check is moved out. But the real logic is still repeated here and in `delete`. What I meant in the comment about is to modularize both the check as well as the snapshot + PVC deletion part.